pkg/actuators/machine/vm.go

```diff
@@ -185,21 +219,64 @@ func deleteVM(ntnxclient *nutanixClientV3.Client, vmUUID string) error {
 	return nil
 }
 
-// findImageByName retrieves the Image with the given vm name
+// findClusterUuidByName retrieves the cluster uuid by the given cluster name
+func findClusterUuidByName(ntnxclient *nutanixClientV3.Client, clusterName string) (*string, error) {
+	klog.Infof("Checking if cluster with name %q exists.", clusterName)
+
+	res, err := ntnxclient.V3.ListCluster(&nutanixClientV3.DSMetadata{
+		//Kind: utils.StringPtr("cluster"),
+		Filter: utils.StringPtr(fmt.Sprintf("name==%s", clusterName)),
+	})
+	if err != nil || len(res.Entities) == 0 {
+		e1 := fmt.Errorf("Failed to find cluster by name %q. error: %v", clusterName, err)
+		klog.Errorf(e1.Error())
+		return nil, e1
+	}
+
+	if len(res.Entities) > 1 {
+		klog.Warningf("Found more than one (%v) clusters with name %q.", len(res.Entities), clusterName)
+	}
+
+	return res.Entities[0].Metadata.UUID, nil
+}
+
+// findImageByName retrieves the image uuid by the given image name
 func findImageUuidByName(ntnxclient *nutanixClientV3.Client, imageName string) (*string, error) {
-	klog.Infof("Checking if Image with name %s exists.", imageName)
+	klog.Infof("Checking if image with name %q exists.", imageName)
 
 	res, err := ntnxclient.V3.ListImage(&nutanixClientV3.DSMetadata{
 		//Kind: utils.StringPtr("image"),
 		Filter: utils.StringPtr(fmt.Sprintf("name==%s", imageName)),
 	})
 	if err != nil || len(res.Entities) == 0 {
-		klog.Errorf("Failed to find image by name %s. error: %v", imageName, err)
-		return nil, fmt.Errorf("Failed to find image by name %s. error: %v", imageName, err)
+		e1 := fmt.Errorf("Failed to find image by name %q. error: %v", imageName, err)
+		klog.Errorf(e1.Error())
+		return nil, e1
+	}
+
+	if len(res.Entities) > 1 {
+		klog.Warningf("Found more than one (%v) images with name %q.", len(res.Entities), imageName)
+	}
+
+	return res.Entities[0].Metadata.UUID, nil
+}
+
+// findSubenetUuidByName retrieves the subnet uuid by the given subnet name
+func findSubenetUuidByName(ntnxclient *nutanixClientV3.Client, subnetName string) (*string, error) {
+	klog.Infof("Checking if subnet with name %s exists.", subnetName)
```

Comment:
i think we should stick with the `%q` convention here as well

Reply:
Changed as suggested.